### ISSUE TITLE
No showed pagination in TAB 

### ISSUE THREAD

**Issue by nng69** [understrap/understrap-child#160]
Hi,

I would like to insert pagination in any TAB. Each TAB show different products for different catergory.
In any TAB I would to see no more than 6 products/page 
So, in TAB where list of products are more than 6, I need to show pagination

I've the following code:

`<?php get_header(); 

function drawProductsPerCategory($cat){
	$paged = ( get_query_var( 'paged' ) ) ? get_query_var( 'paged' ) : 1;
    $args = array( 'post_type' => 'product', 'posts_per_page' => 6, 'product_cat' => $cat, 'paged' => $paged );
    $loop = new WP_Query( $args );

    while ( $loop->have_posts() ) : $loop->the_post(); 
    global $product; 

    echo '<div id="product-'.$product->get_id().'" class="config-item" data-product-price="'.$product->get_price().'" data-product-id="'.$product->get_id().'" data-product-category="'.$cat.'" ><img class="product-img" src="';
    the_post_thumbnail_url(300);
    echo '" alt="'.get_the_title().'" /></div>';
    endwhile;

	
 wp_reset_query(); 
	
	
	
	
	
	
}

/*
function drawProductsImages($cat){
    
    $args = array( 'post_type' => 'product', 'posts_per_page' => 100, 'product_cat' => $cat );
    $loop = new WP_Query( $args );

    while ( $loop->have_posts() ) : $loop->the_post(); 
    global $product; 
    
    // echo '<img id="component-'.$product->get_id().'" class="component-img component-'.$cat.'" src="http://amigala.com/wp-content/themes/storefront/assets/components/'.$product->get_id().'.png" alt="'.get_the_title().'" />';
    echo '<div id="component-'.$product->get_id().'" class="component-img component-'.$cat.'"></div>';

    endwhile; 

    wp_reset_query(); 
}
*/

?>

		
<section>

<?php /* if (have_posts()) : while (have_posts()) : the_post(); ?>
    <h1 style="text-align: center;"><?php the_title(); ?></h1>
    <div style="text-align: center;"><?php the_content(); ?></div>
    <?php do_action( 'storefront_page_after' ); ?>
    <?php endwhile; else: ?>
    <div><?php _e('<h2>Oops...</h2><br>I couldn\'t find what you were looking for.'); ?></div>
<?php endif; */ ?>

</section>

<style>
    .config-col{
        width:100%; float: left; text-align: center;
    }
    .config-tit{
        text-align: left;
    }
    .config-border{
        border: 2px solid #333;
        margin-top: 30px;
        margin-bottom: 60px;
        padding: 15px;
    }
    .config-tab{
        width: 18%; margin: 10px 1% 20px; float: left; cursor: pointer;
        border: 1px solid #333;
        padding-top: 5px;
        padding-bottom: 5px;
    }
    .config-tab.current{
        background-color: transparent;
    }
    .config-tab.current h4{
        color: #333333 !important;
    }
    .config-tab h4{
        font-size: 14px;
        margin-bottom: 0;
    }
    .config-tab:hover{
        background-color: #eee !important;
    }
    .config-tab:hover h4{
        color: #333333 !important;
    }
    .config-list{
        clear: both;
    }
    .config-item{
        width: 31%; margin: 1%; float: left; cursor:pointer;
    }
    .config-item img{
        width: 100%;
    }
    .config-check{
        margin-top: -100%;
    }
    .config-item .config-check img{
        visibility: hidden;
    }
    .config-item.selected .config-check img, .config-item:hover .config-check img{
        visibility: visible;
    }
    .config-item.selected .product-img, .config-item:hover .product-img{
        opacity: 0.4;
    }
    .config-total{
        color: white;
        background: #333;
        clear: both;
        padding: 10px 15px 12px;
        text-align: left;
    }
    .config-total p{
        margin: 0;
    }
    .config-small{
        font-size: 12px;
        color: #aaa;
        margin-top: -5px;
    }
    .config-buy{
        float: right;
        width: 90px;
        background-color: #ffffff;
        color: #333333;
        text-align: center;
        height: 30px;
        margin-top: -35px;
        font-size: 14px;
        line-height: 30px;
        cursor: pointer;
    }
    .config-buy:hover{
        opacity: 0.9;
    }
    .config-buy.disabled{
        opacity: 0.5 !important;
        cursor:not-allowed;
    }
    
    .component-box{
        width: 100%;
        height: auto;
        overflow: hidden;
        min-height: 600px;
        max-width: 600px;
        position: relative;
        margin: 0 auto 30px;
    }
    
    .component-img{
        position: absolute;
        top: 0;
        left: 0;
        width: 100%;
        height: auto;
    }
    
    .component-interiores{
        z-index: 10;
        margin-top: -4px;
    }
    .component-bases{
        z-index: 20;
    }
    .component-solapas{
        z-index: 30;
    }
    .component-asas{
        z-index: 40;
    }
    .component-correas{
        z-index: 50;
    }
        
    @media screen and (min-width: 992px) {
        .config-col{
            width:50%; 
        }
    }
</style>

<section class="container">

    <br><br>

    <div class="row">

        <div class="config-col">
            <div class="component-box">
            <img class="component-img component-bases" src="http://amigala.com/wp-content/themes/storefront/assets/components/default-bases.png" alt="Bases" />
            <img class="component-img component-asas" src="http://amigala.com/wp-content/themes/storefront/assets/components/default-asas.png" alt="Asas" />
            <img class="component-img component-interiores" src="http://amigala.com/wp-content/themes/storefront/assets/components/default-interiores.png" alt="Interiores" />
            <img class="component-img component-solapas" src="http://amigala.com/wp-content/themes/storefront/assets/components/default-solapas.png" alt="Solapas" />
            <img class="component-img component-correas" src="http://amigala.com/wp-content/themes/storefront/assets/components/default-correas.png" alt="Correas" />
            <?php /* drawProductsImages('bases'); ?>
            <?php drawProductsImages('asas'); ?>
            <?php drawProductsImages('interiores'); ?>
            <?php drawProductsImages('solapas'); ?>
            <?php drawProductsImages('correas'); */ ?>
            </div>
        </div>

        <div class="config-col">
            
            <div class="config-tit">
                <h2>Diseña tu bolso ideal</h2>
                <p>¡Customiza tu bolso para tu ocasión! Elige la base, las asas y los demás elementos.</p>
            </div>
            
            <div class="config-border">
            
                <div class="config-tab" id="bases-tab" onclick="showCatList('#bases')">
                    <h4>Base *</h4>
                </div>
                <div class="config-tab" id="asas-tab" onclick="showCatList('#asas')">
                    <h4>Asas *</h4>
                </div>
                <div class="config-tab" id="interiores-tab" onclick="showCatList('#interiores')">
                    <h4>Interior</h4>
                </div>
                <div class="config-tab" id="solapas-tab" onclick="showCatList('#solapas')">
                    <h4>Solapa</h4>
                </div>
				
				
				
                
                <div class="config-tab" id="correas-tab" onclick="showCatList('#correas')">
                    <h4>Correa</h4>
                </div>               
				
                <div id="bases" class="config-list">
                    <?php drawProductsPerCategory('bases'); ?>
					
					
                </div>              

                <div id="asas" class="config-list">
                    <?php drawProductsPerCategory('asas'); ?>
                </div>

                <div id="interiores" class="config-list">
                    <?php drawProductsPerCategory('interiores'); ?>
                </div> 

                 <div id="solapas" class="config-list">
                    <?php drawProductsPerCategory('solapas'); ?>
                </div> 

                
                 <div id="correas" class="config-list">
                    <?php drawProductsPerCategory('correas'); ?>
                </div> 
                
                
                <div class="config-total">
                    <p><b><span id="totalprice">0</span>€</b></p>
                    <p class="config-small">(Incluye IVA, no incluye envío)</p>
                    <div class="config-buy disabled">comprar</div>
                </div>
                <br>
            
            </div>
        </div>

    </div>

    <br><br>

</section>

<?php /* <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.6.5/angular.min.js"></script> */ ?>

<script>    
    
    /*
    function resizeBgComponents(){
        var w = jQuery('.component-img').width();
        var r = w/600;
        jQuery('.component-img').css('background-size', r*12552+'px '+r*3000+'px');
    }
    */
   
    function showCatList(id){
        jQuery('.config-list').hide();
        jQuery(id).show();
        jQuery('.config-tab').removeClass('current');
        jQuery(id+'-tab').addClass('current');
    }
    
    function calcTotalPrice(){
        var a = jQuery('.config-item.selected');
        var t = 0;
        var bases = false;
        var asas = false;
        
        for (var i = 0; i < a.length; ++i) {
            t = t + parseFloat(a[i].dataset.productPrice);
            var c = a[i].dataset.productCategory;
            if(c === 'bases'){
                bases = true;
            }
            if(c === 'asas'){
                asas = true;
            }
        }
        jQuery('#totalprice').text(t);
        if(bases && asas){
            jQuery('.config-buy').removeClass('disabled');
        }else{
            jQuery('.config-buy').addClass('disabled');
        }
    }
    
    /*
    function postData(data,endpoint)
    {
        return new Promise(function(resolve, reject) {
            jQuery.ajax({
                type: "POST",
                crossDomain: true,
                url: endpoint,
                data: data,
                dataType : "json",
                contentType : "application/x-www-form-urlencoded"
            })
            .done(resolve)
            .fail(reject);
        });
    }
    */
   
    window.addEventListener('load', function() {
        showCatList('#bases');
        jQuery('.config-item').append('<div class="config-check"><img src="http://amigala.com/wp-content/themes/storefront/assets/images/check-color.png" alt="Seleccionado" /></div>');
        jQuery('body').on('click','.config-item', function() {
            var id = jQuery(this).attr('data-product-id');
            var cat = jQuery(this).attr('data-product-category');
            
            // jQuery('.component-'+cat).hide();
            
            if(jQuery(this).hasClass('selected')){
                jQuery(this).removeClass('selected');
                // jQuery('#default-'+cat).show();
                jQuery('.component-'+cat).attr('src','http://amigala.com/wp-content/themes/storefront/assets/components/default-'+cat+'.png');
            }else{
                jQuery('#'+cat+' .config-item').removeClass('selected');
                jQuery(this).addClass('selected');
                // jQuery('#component-'+id).show();
                jQuery('.component-'+cat).attr('src','http://amigala.com/wp-content/themes/storefront/assets/components/'+id+'.png');
            }
            
            calcTotalPrice();
        });
        
        jQuery('body').on('click','.config-buy', function() {
            
            if(jQuery(this).hasClass('disabled')){
                // do nothing
                return false;
            }
            
            var a = jQuery('.config-item.selected');
            
            var url = 'http://amigala.com/cart/?add-to-cart-multiple=';
            
            for (var i = 0; i < a.length; ++i) {
                var id = a[i].dataset.productId;
                url = url+id+',';
            }
            
            url = url.slice(0, -1);
            window.location.href = url;
           
            /*
            for (var i = 0; i < a.length; ++i) {
               
                var sendData = {
                    product_sku : '',
                    product_id : a[i].dataset.productId,
                    quantity : 1
                };

                postData(sendData,'http://amigala.com/cart/?wc-ajax=add_to_cart').then(function(response){
                    console.log(response);
                });
            }
            */
            
        });
        
        /*
        resizeBgComponents();
        
        window.onresize = function() {
            resizeBgComponents();
        };
        */
    });
</script>

<?php
get_footer();`

I have try different solution but I'm not able to show pagination.

So need your help. Thanks in advance.

JK

**Comment by Thomas-A-Reinert** [comment]
Sorry, but none of your functions is a WordPress/WooCommerce/UnderStrap default function. And except of [pagination](https://getbootstrap.com/docs/4.0/components/pagination/) and [tabs](https://getbootstrap.com/docs/4.0/components/navs/#tabs) there is no connection to UnderStrap. So that must be some self-written/third party code. And it´s in totally _ugly_ format jumping between code-format and non-code-format, compressed code etc. so I cannot even understand it easily - sorry.


I don´t have a code to reproduce or download, neither a link. So, I´m sorry. It´s not easily reproducible therefore I gotta quit.

I´ll also close it because it doesn´t seem UnderStrap-related to me in any case.
